src/aotcompile.cpp

```diff
@@ -957,8 +957,9 @@ static void add_output_impl(Module &M, TargetMachine &SourceTM, std::string *out
         AnalysisManagers AM{*TM, PB, OptimizationLevel::O0};
         ModulePassManager MPM;
         MPM.addPass(BitcodeWriterPass(OS));
-        *unopt = NewArchiveMember(MemoryBufferRef(*outputs, names[0]));
-        outputs++;
+        outputs[1] = (names[0] + "#" + std::to_string(shardidx) + "_unopt.bc").str();
```

Comment:
This will still make the corrupt string `data_unopt.bc#1_unopt.bc` though, right? The `names[0]` should not be adding the extension twice.